### PR TITLE
feat: add free platform plan

### PR DESCRIPTION
**Pull request by ThyMinimalDev**
## What does this PR do?

Replace Platform Starter Plan with a new Free plan

Screenshot 2024-11-11 at 17 29 21

also mark payment as overdue when payment fails and cancel subscription when deleted by stripe 

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

### CODE REVIEW

File: apps/api/v2/src/modules/billing/controllers/billing.controller.ts [hunk 1]
```diff
@@ -108,7 +108,22 @@ export class BillingController {
       this.stripeWhSecret
     );
 
-    await this.billingService.createOrUpdateStripeSubscription(event);
+    switch (event.type) {
+      case "checkout.session.completed":
+        await this.billingService.handleStripeCheckoutEvents(event);
```
Comment (ThyMinimalDev) [inline apps/api/v2/src/modules/billing/controllers/billing.controller.ts:113]: subscribe / upgrade

File: apps/api/v2/src/modules/billing/controllers/billing.controller.ts [hunk 2]
```diff
@@ -108,7 +108,22 @@ export class BillingController {
...
+        break;
+      case "customer.subscription.deleted":
+        await this.billingService.handleStripeSubscriptionDeleted(event);
```
Comment (ThyMinimalDev) [inline apps/api/v2/src/modules/billing/controllers/billing.controller.ts:116]: stripe deleted subscription after multiple failed payment retries

File: apps/api/v2/src/modules/billing/controllers/billing.controller.ts [hunk 3]
```diff
@@ -108,7 +108,22 @@ export class BillingController {
...
+        break;
+      case "invoice.payment_failed":
+        await this.billingService.handleStripePaymentFailed(event);
```
Comment (ThyMinimalDev) [inline apps/api/v2/src/modules/billing/controllers/billing.controller.ts:119]: payment failed, mark subscription as overdue

File: apps/api/v2/src/modules/billing/controllers/billing.controller.ts [hunk 4]
```diff
@@ -108,7 +108,22 @@ export class BillingController {
...
+        break;
+      case "invoice.payment_succeeded":
+        await this.billingService.handleStripePaymentSuccess(event);
```
Comment (ThyMinimalDev) [inline apps/api/v2/src/modules/billing/controllers/billing.controller.ts:122]: payment success, mark subscription as not overdue

**Review by supalarry — APPROVED** [review]

File: apps/api/v2/src/modules/billing/types.ts [hunk 5]
```diff
@@ -1,8 +1,9 @@
 export enum PlatformPlan {
+  FREE = "FREE",
   STARTER = "STARTER",
```
[the remark this answers is not in the document]
Comment (ThyMinimalDev) [reply apps/api/v2/src/modules/billing/types.ts:3]: not while we have people still using this plan

**Pull request merged**